lib/CoreBot.js

```diff
@@ -568,10 +576,13 @@ function Botkit(configuration) {
                         (duration > this.task.timeLimit) && // timelimit is up
                         (lastActive > this.task.timeLimit) // nobody has typed for 60 seconds at least
                     ) {
-
-                        if (this.threads.timeout) {
+                        // if timeoutHandler is set then call it, otherwise follow the normal flow
+                        // this will not break others code, after the update
+                        if (this.timeOutHandler != null || this.timeOutHandler != undefined) {
```

Comment:
Why would you want to check the type of a variable? I didn't understand. In JS `null` or `undefined` is transformed into `false` in logical operator check. Therefore `if (this.timeOutHandler)` will cover both of the cases in this if statement.